alacritty_terminal/src/tty/windows/mod.rs

```diff
@@ -60,7 +58,52 @@ pub struct Pty {
 
 impl Pty {
     pub fn resize_handle(&self) -> impl OnResize {
-        self.handle.clone()
+        match &self.handle {
+            PtyHandle::Winpty(w) => PtyResizeHandle::Winpty(w.resize_handle()),
+            PtyHandle::Conpty(c) => PtyResizeHandle::Conpty(c.resize_handle()),
+        }
+    }
+
+    fn new(handle: PtyHandle, child_watcher: ChildExitWatcher) -> Self {
+        Self {
+            handle,
+            read_token: 0.into(),
+            write_token: 0.into(),
+            child_event_token: 0.into(),
+            child_watcher,
+        }
+    }
+
+    #[inline]
+    fn conout(&self) -> &EventedReadablePipe {
+        match &self.handle {
+            PtyHandle::Winpty(w) => &w.conout,
+            PtyHandle::Conpty(c) => &c.conout,
+        }
+    }
+
+    #[inline]
+    fn conout_mut(&mut self) -> &mut EventedReadablePipe {
+        match &mut self.handle {
+            PtyHandle::Winpty(w) => &mut w.conout,
+            PtyHandle::Conpty(c) => &mut c.conout,
+        }
+    }
+
+    #[inline]
+    fn conin(&self) -> &EventedWritablePipe {
+        match &self.handle {
+            PtyHandle::Winpty(w) => &w.conin,
+            PtyHandle::Conpty(c) => &c.conin,
+        }
+    }
+
+    #[inline]
+    fn conin_mut(&mut self) -> &mut EventedWritablePipe {
+        match &mut self.handle {
+            PtyHandle::Winpty(w) => &mut w.conin,
+            PtyHandle::Conpty(c) => &mut c.conin,
+        }
```

Comment:
All of this would be gone if we keep the `conin` and `conout` on the `PTY`. This code is completely useless and just costs us maintenance burden.